src/main/java/com/googleintern/wfm/ruleengine/action/CsvWriter.java

```diff
@@ -0,0 +1,107 @@
+package src.main.java.com.googleintern.wfm.ruleengine.action;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.opencsv.CSVWriter;
+import src.main.java.com.googleintern.wfm.ruleengine.model.FilterModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.RuleModel;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+/** CsvWriter class is used to write newly generated rules into a csv file. */
+public class CsvWriter {
+  enum Separator {
+    SEMICOLON(";"),
+    COMMA(","),
+    SQUARE_BRACKET_LEFT("["),
+    SQUARE_BRACKET_RIGHT("]"),
+    CURLY_BRACKET_LEFT("{"),
+    CURLY_BRACKET_RIGHT("}");
+
+    final String symbol;
+
+    Separator(String symbol) {
+      this.symbol = symbol;
+    }
+  }
+
+  private static final String SKILL_ID_PREFIX = "skill_id:";
+  private static final String ROLE_ID_PREFIX = "role_id:";
+  private static final String[] CSV_FILE_HEADER =
+      new String[] {
+        "Workforce ID", "Workgroup ID", "Case Pool ID", "Permission Set IDs", "Filters"
+      };
+
+  /**
+   * Create a new csv file using OUTPUT_CSV_FILE_NAME as name in outputCsvFileLocation and write
+   * data into the newly created csv file.
+   *
+   * @throws IOException
+   */
+  public static void writeDataIntoCsvFile(String outputCsvFilePath, ImmutableList<RuleModel> rules)
+      throws IOException {
+    File outputFile = new File(outputCsvFilePath);
+    Files.deleteIfExists(outputFile.toPath());
+    outputFile.createNewFile();
+
+    FileWriter outputFileWriter = new FileWriter(outputFile);
+    CSVWriter csvWriter = new CSVWriter(outputFileWriter);
+
+    csvWriter.writeNext(CSV_FILE_HEADER);
+    ImmutableList<String[]> data =
+        rules.stream().map(CsvWriter::writeData).collect(ImmutableList.toImmutableList());
+
+    csvWriter.writeAll(data);
+    csvWriter.close();
+  }
+
+  private static String[] writeData(final RuleModel rule) {
```

Comment:
would rename this to convertRuleToRow or similar since this is not writing data, but converting to a row for writing